.github/docs-pr-preview.yml

```diff
@@ -0,0 +1,49 @@
+name: Docs PR Preview
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+    paths:
+      - 'docs/**'      
+      
+
+permissions:
+  contents: write       
+  pull-requests: write   # To comment on the PR automatically
+
+jobs:
+  build-and-deploy-preview:
+    runs-on: ubuntu-latest
+
+    steps:
+      #  Check out the PR code
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Setup Python (matches your docs environment)
+      - name: Setup Python 3.x
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      # Install dependencies inside virtual environment
+      - name: Install dependencies
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          pip install -r docs/requirements.txt
+
+      #  Build documentation site
+      - name: Build documentation
+        run: |
+          source venv/bin/activate
+          mkdocs build --site-dir build/
+
+      # Deploy preview to gh-pages branch under pr-preview path
+      - name: Deploy preview
+        uses: rossjrw/pr-preview-action@v1
```

Comment:
Does this Action really work correctly with the structure that `mike` maintains?

Reply:
Reading the documentation for that Action, I doubt it.

Since we already have automation using `mike`, I think that the simplest thing would be to use `mike` consistently.